enhancements/installer/aws-load-balancer-type.md

```diff
@@ -0,0 +1,343 @@
+  //   the transport layer (TCP/SSL) or the application layer (HTTP/HTTPS). See
+  //   the following for additional details:
+  //   https://docs.aws.amazon.com/AmazonECS/latest/developerguide/load-balancer-types.html#clb
+  //
+  // * "NLB": A Network Load Balancer that makes routing decisions at the
+  //   transport layer (TCP/SSL). See the following for additional details:
+  //   https://docs.aws.amazon.com/AmazonECS/latest/developerguide/load-balancer-types.html#nlb
+  // +unionDiscriminator
+  // +kubebuilder:validation:Enum:=NLB;Classic
+  // +kubebuilder:validation:Required 
+  Type AWSLBType `json:"type,omitempty"`
+}
+
+type AWSLBType string
+
+const (
+    // NLB is the Network Load Balancer Type of AWS. Using NLB one can set NLB load balancer type for the default ingress controller.
+    NLB AWSLBType = "NLB"
+
+    // Classic is the Classic Load Balancer Type of AWS. Using CLassic one can set Classic load balancer type for the default ingress controller.
+    Classic AWSLBType = "Classic"
+)
+```
+
+#### Resource provided to the installer
+
+The users provide a load balancer type to the installer.
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  aws:
+    region: us-west-2
+    lbType: nlb
+    subnets:
+    - subnet-1
+    - subnet-2
+    - subnet-3
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+
+#### Resources created by the installer
+
+- The installer will create infrastructure cluster object with the load balancer type defined in it.
+
+```yaml
+ oc edit ingress.config/cluster -o yaml
+ # Please edit the object below. Lines beginning with a '#' will be ignored,
+ # and an empty file will abort the edit. If an error occurs while saving this file will be
+ # reopened with the relevant failures.
+ #
+ apiVersion: config.openshift.io/v1
+ kind: Ingress
+ metadata:
+   creationTimestamp: "2022-08-30T01:03:17Z"
+   generation: 11
+   name: cluster
+   resourceVersion: "60888"
+   uid: de89cc95-449e-4ae5-a837-f4171fe61e36
+ spec:
+   domain: apps.testnlbapi.devcluster.openshift.com
+   loadbalancer:
+     platform:
+       aws:
+         type: NLB
+       type: AWS
+ status:
+   componentRoutes:
+```
+
+- The cluster-ingress-operator will check the cluster ingress object for the lbType and reconcile to create an default ingress controller
+  with the specified lbType.
+
+#### Destroying cluster
+
+- None
+
+
+### Risks and Mitigations
+
+- Users need to put either nlb or elb as input or else the LB creation will fail.
+
+- To mitigate this we need to validate the input.
+
+### Drawbacks
+
+None
+
+## Design Details
+
+### Test Plan
+
+- Set the awsLBType field in the install-config.yaml either to nlb or elb and see if the default ingress controller gets created with the desired LB type.
+- Delete the default ingress controller CR and see if the default ingress controller will be created using the LB type mentioned during installation.
```

Comment:
Are these going to be E2E tests in openshift/installer or what?

Reply:
Manual test.

Reply:
BTW I have already written a unit test case in the installer as well.